tests/unit/anchore_engine/clients/test_grype_wrapper.py

```diff
@@ -100,12 +101,53 @@ def test_get_missing_grype_db_session():
         # Function under test
         grype_wrapper_singleton._grype_db_session_maker
 
+    # Validate error message
     assert (
         str(error.value)
         is GrypeWrapperSingleton.MISSING_GRYPE_DB_SESSION_MAKER_ERROR_MESSAGE
     )
 
 
+def test_get_current_grype_db_checksum(grype_db_dir):
+    # Create grype_wrapper_singleton instance
+    grype_wrapper_singleton = TestGrypeWrapperSingleton.get_instance()
+
+    # Setup test input
+    grype_wrapper_singleton._grype_db_dir = grype_db_dir
+
+    # Function under test
+    result = grype_wrapper_singleton.get_current_grype_db_checksum()
+
+    # Validate result
+    assert result == NEW_VERSION_NAME
+
+
+def test_get_current_grype_db_checksum_missing_db_dir_value():
+    # Create grype_wrapper_singleton instance
+    grype_wrapper_singleton = TestGrypeWrapperSingleton.get_instance()
+
+    # Setup test input with a non-existant path
+    grype_wrapper_singleton._grype_db_dir = "/does/not/exist"
+
+    # Function under test
+    result = grype_wrapper_singleton.get_current_grype_db_checksum()
+
+    # Validate error message
+    assert result is None
+
+
+def test_get_current_grype_db_checksum_missing_db_dir():
+    # Create grype_wrapper_singleton instance
+    grype_wrapper_singleton = TestGrypeWrapperSingleton.get_instance()
+
+    with pytest.raises(ValueError) as error:
+        # Function under test
+        result = grype_wrapper_singleton.get_current_grype_db_checksum()
+
+    # Validate error message
+    assert str(error.value) == GrypeWrapperSingleton.MISSING_GRYPE_DB_DIR_ERROR_MESSAGE
```

Comment:
pytest.raises will actually throw an exception if the expected exception isn't raised. Is the second assertion necessary? if you're writing these checks, it may be a sign that it's time to implement custom exception classes for the grype wrapper.

https://docs.pytest.org/en/6.2.x/reference.html#pytest-raises

Reply:
More of a nicety to confirm the exception message.